Utils/azuretools-core/src/com/microsoft/azuretools/azurecommons/util/GetHashMac.java

```diff
@@ -42,38 +53,40 @@ public static boolean IsValidHashMacFormat(String hashMac) {
         return matcher.matches();
     }
 
-    public static String GetHashMac() {
+    public static String getHashMac() {
         String ret = null;
-        String mac_raw = GetRawMac();
-        if (mac_raw != null && !mac_raw.isEmpty()) {
-            String mac_regex = "([0-9A-Fa-f]{2}[:-]){5}[0-9A-Fa-f]{2}";
+        String mac_raw = getRawMac();
+        mac_raw = isValidMac(mac_raw) ? mac_raw : getRawMacWithoutIfconfig();
```

Comment:
The return might by empty string. How to handle this?

Reply:
Yes, we may lose the use if we can't get the mac with IP tools or Java NetworkInterface, but I think it's fine as there will be few users will meet this issue.

Reply:
How about to generate an UUID and put it into the `<user home>/AzureToolsForIntelliJ` folder?

Reply:
@wezhang It should work to generate a UUID, but I am not sure how many users will meet this issue, they could send telemetries but we can't get the mac address. Besides, if we want to do this job, we may better to this in plugin initialization functions, for IntelliJ is [here](https://github.com/microsoft/azure-tools-for-java/blob/develop/PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/AzurePlugin.java#L155). If get mac returns empty, generate a hash UUID installation id.